jb/itch5/array_based_order_book.hpp

```diff
@@ -22,14 +21,16 @@ namespace itch5 {
 
 // forward declaration of
 void validate_add_order_params(int, price4_t px = price4_t(0));
+void handle_exception(std::string, std::size_t, price4_t, int);
+void handle_exception(
+    std::string, std::size_t, std::size_t, std::size_t, price4_t, int);
 
 namespace defaults {
 
 #ifndef JB_ARRAY_DEFAULTS_max_size
 #define JB_ARRAY_DEFAULTS_max_size 10000
 #endif
-
-std::size_t max_size = JB_ARRAY_DEFAULTS_max_size;
+static std::size_t Max_Size = JB_ARRAY_DEFAULTS_max_size;
```

Comment:
> BTW< what is this?

A macro that defines the constructors for a config object.  They are always the same but have to be explicitly defined, I know it is yucky.

> What did you notice?

I thought I was missing some of the changes I had requested in the previous review.